redfish/common.go

```diff
@@ -118,14 +118,23 @@ func PowerOperation(resetType string, maximumWaitTime int, checkInterval int, se
 		}
 	}
 
-	if resetType == "ForceRestart" || resetType == "GracefulRestart" || resetType == "PowerCycle" {
+	if resetType == "ForceRestart" || resetType == "GracefulRestart" || resetType == "PowerCycle" || resetType == "Nmi" {
 		// If someone asks for a reset while the server is off, change the reset type to on instead
 		if system.PowerState == "Off" {
 			resetType = "On"
 		}
 		targetPowerState = "On"
 	}
 
+	if resetType == "PushPowerButton" {
```

Comment:
we can make it **else if** here, and in the above condition we can do it by
`if resetType in ["ForceRestart", "GracefulRestart", "PowerCycle", "Nmi"]:` for better understanding

Reply:
you can do it in the future not a big deal.